pdns/dnsdist.hh

```diff
@@ -71,6 +196,7 @@ struct DNSQuestion
   bool skipCache{false};
   bool ecsOverride;
   bool useECS{true};    
+  QTag qTag;                        // GCA - Seth Ornstein - extra class for tags 5/30/2017
```

Comment:
Since we are trying to keep the `DNSQuestion` object as light as possible for performance reason, would you be willing to consider making that an `std::unique_ptr<QTag>` so that we don't pay the cost of constructing the object until we really use it? Please also move it up in the `DNSQuestion` object with the other pointers to make padding more optimal.
By the way, if I'm not mistaken, the `QTag` object is basically a `std::unordered_map<std::string,std::string>`, perhaps we could avoid defining a new object and just handle the needful in the `Lua` bindings?